### ISSUE TITLE
App with a ListView crashes on repeated orientation changes in iOS simulator

### ISSUE THREAD

**Issue by mikekreuzer** [flutter/flutter#15438]
## Steps to Reproduce

- Take an app with a ListView (eg from the [write your first flutter app](https://gist.githubusercontent.com/Sfshaza/936ce0059029a8c6e88aaa826a3789cd/raw/a3065d5c681a81eff32f75a9cd5f4d9a5b24f9ff/main.dart))
- add some text concatenation (I'm unsure if this is the problem or if this just speeds things along), eg make line 57 ````pair.asPascalCase + 'do some concatting',````
- run the app, rotate the hardware in the sim a few times this way & that, and... crash.

## Logs

Terminal just says 'Lost connection to device' but if I run this in Xcode the app crashes with EXC_I386_GPFLT

May be related to #13300 and #6995

**Comment by pulyaevskiy** [comment]
Seems like I started getting the same error today after I added some animations to my project. Symptoms are very similar except instead of rotation the app exits when keyboard pops up while there is an animation in progress (I was able to reproduce with scale animations).

## Steps to Reproduce

#### 1. Create "increment counter" app:

```shell
flutter create scale_bug
```
#### 2. Update source code

Go to `scale_bug/lib/main.dart` and replace `_MyHomePageState` class with slightly adjusted version in this Gist: https://gist.github.com/pulyaevskiy/e266e70cbf71c31aaa6a14ecd7cf0a2a
(Adds a TextField to the Scaffold and when it gains focus forces FAB disappear with "scale down" animation).

#### 3. Run the app (make sure to do clean as well)

```shell
open -a Simulator.app
flutter clean
flutter run
```

#### 4. Tap on the `TextField` 

So that it forces on screen keyboard to slide up.

Result: the app crashes with nothing in logs and "Lost connection to device" in the terminal. Running in Xcode shows error code `EXC_I386_GPFLT`.

**Comment by pulyaevskiy** [comment]
Just got this issue again when using BottomAppBar with a docked FAB. Really annoying when working on a screen with a text input.

**Comment by pulyaevskiy** [comment]
Tried to debug this by copy-pasting FAB animator and tweaking it. Looks like disabling rotation animation makes this issue go away, so main culprit at this point is `RotationTransition` widget.

**Comment by jason-simmons** [comment]
Captured a stack trace of a zombie object in AccessibilityBridge::UpdateSemantics

@chinmaygarde 

```
 	Malloc/Autorelease (2)	+1	 	00:57.692.599	Flutter	shell::AccessibilityBridge::UpdateSemantics(std::__1::unordered_map<int, blink::SemanticsNode, std::__1::hash<int>, std::__1::equal_to<int>, std::__1::allocator<std::__1::pair<int const, blink::SemanticsNode> > >)
0	 Malloc	+1	1	00:57.692.599	Flutter	shell::AccessibilityBridge::UpdateSemantics(std::__1::unordered_map<int, blink::SemanticsNode, std::__1::hash<int>, std::__1::equal_to<int>, std::__1::allocator<std::__1::pair<int const, blink::SemanticsNode> > >)
1	 Autorelease		 	00:57.692.600	Flutter	shell::PlatformViewIOS::UpdateSemantics(std::__1::unordered_map<int, blink::SemanticsNode, std::__1::hash<int>, std::__1::equal_to<int>, std::__1::allocator<std::__1::pair<int const, blink::SemanticsNode> > >)
2	Retain	+1	2	00:57.692.605	Flutter	shell::AccessibilityBridge::UpdateSemantics(std::__1::unordered_map<int, blink::SemanticsNode, std::__1::hash<int>, std::__1::equal_to<int>, std::__1::allocator<std::__1::pair<int const, blink::SemanticsNode> > >)
3	Zombie		-1	00:57.693.663	Flutter	fml::MessageLoopDarwin::OnTimerFire(__CFRunLoopTimer*, fml::MessageLoopDarwin*)
```

**Comment by jason-simmons** [comment]
@goderbauer

**Comment by calmh** [comment]
Seeing same with just a `CustomScrollView` at the top level with a `SliverAppBar` inside. Rotate => Crash.

**Comment by corymalcolmtaylor** [comment]
I am getting this issue using the code from the Flutter Udacity course [here](https://github.com/flutter/udacity-course/tree/master/course/08_responsive/solution_08_responsive)

**Comment by goderbauer** [comment]
FYI @jonahwilliams Looks like the crash is related to a11y (see @jason-simmons  comment).

**Comment by jonahwilliams** [comment]
I've managed to reproduce this once or twice out of hundreds of orientation changes. While I agree the cause should be found and fixed, I don't have any time to work on it right now

**Comment by GaryQian** [comment]
Is this reproducible on iOS physical devices?

Also, it has been a long time since the last update, does anyone still experience this bug?
